test/core/security/grpc_tls_crl_provider_test.cc

```diff
@@ -64,8 +61,41 @@ using ::grpc_core::experimental::StaticCrlProvider;
 namespace grpc_core {
 namespace testing {
 
+absl::StatusOr<std::shared_ptr<CrlProvider>>
+CreateDirectoryReloaderCrlProviderForTest(
+    absl::string_view directory, std::chrono::seconds refresh_duration,
+    std::function<void(absl::Status)> reload_error_callback,
+    std::shared_ptr<grpc_event_engine::experimental::EventEngine> event_engine,
+    std::shared_ptr<Directory> directory_impl) {
+  if (directory_impl == nullptr) {
+    directory_impl = std::make_shared<Directory>(directory);
+  }
+  if (event_engine == nullptr) {
+    event_engine = grpc_event_engine::experimental::GetDefaultEventEngine();
+  }
+
+  auto provider = std::make_shared<experimental::DirectoryReloaderCrlProvider>(
+      refresh_duration, reload_error_callback, event_engine, directory_impl);
+  absl::Status initial_status = provider->Update();
+  if (!initial_status.ok()) {
+    return initial_status;
+  }
+  provider->ScheduleReload();
+  return provider;
+}
+
+class DirectoryForTest : public Directory {
+ public:
+  DirectoryForTest() : Directory("") {}
+  ~DirectoryForTest() override = default;
+  absl::StatusOr<std::vector<std::string>> GetFilesInDirectory() override {
```

Comment:
We also need a way to trigger this to return a not-OK status, so that we can make sure the provider behaves correctly in that case.

Reply:
Done and added a test